01_fib/undrewb/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fib(7)
+}
+
+func abs(x int) int {
+	if x < 0 {
+		return -x
+	}
+	return x
+}
+
+func fibonacciSign(n int) int {
+	if n < 0 {
+		return [2]int{1, -1}[abs(n+1)%2]
+	}
+	return 1
+}
+
+func fib(n int) {
+	if n > 20 || n < -20 {
+		panic("inconceivable")
```

Comment:
You keep using that word. I do not think it means what you think it means.

Reply:
I can tell you that fib(9000000) is 1,880,889 digits long from using `math/big.Int`, so I think I could conceive of a number outside this range.

But more seriously, consider whether `panic` is appropriate here. `panic` in Go is used sparingly - in particular you should not make panic a part of a package's API. In this case you're not making a package but assuming the more general implementation, `fib` could easily be part of some `math` package. In that case, it should either return an error or print out the error the same way it prints out the results. Having the caller handle the panic is un-Go-like.

More details here: https://github.com/golang/go/wiki/PanicAndRecover#usage-in-a-package

Reply:
Makes sense cam. Originally I had a print but changed it on a reviewers suggestion. The limit is arbitrary and I agree it could be bigger. I'll make it an error since I can see that panic is a bit over the top.

Reply:
To be clear:
> Better to return an `error` or `panic`.